### ISSUE TITLE
npm state ignore the requested version

### ISSUE THREAD

**Issue by micaelbergeron** [saltstack/salt#23343]
I am having a problem with the npm state, as the requested version isn't getting installed. I end up having the latest version of the node module.

Here is my sls:
**redis-commander.sls**
```
redis-commander@0.2.1:
  npm.installed:
    - require:
      - pkg: npm
```
**Debug output**
```
[INFO    ] Executing state npm.installed for redis-commander@0.2.1
[INFO    ] Executing command 'npm --version' in directory '/root'
[INFO    ] Executing command 'npm list --silent --json --global' in directory '/root'
[INFO    ] Executing command 'npm --version' in directory '/root'
[INFO    ] Executing command 'npm install --silent --json --global "redis-commander"' in directory '/root'
```

**Highstate output**
```
          ID: redis-commander@0.2.1
    Function: npm.installed
      Result: True
     Comment: Package(s) 'redis-commander@0.2.1' successfully installed
     Started: 19:35:45.635635
    Duration: 228559.429 ms
     Changes:
              ----------
              new:
                  - redis-commander@0.2.1
              old:
```

**$ sudo salt-call --local npm.list redis-commander**
```
local:
    ----------
    redis-commander:
        ----------
        from:
            redis-commander@
        version:
            0.3.0
```

However, using the npm module directly (salt '*' npm.install redis-commander@0.2.1)  works fine.
```
local:
    ----------
    redis-commander:
        ----------
        from:
            redis-commander@0.2.1
        resolved:
            https://registry.npmjs.org/redis-commander/-/redis-commander-0.2.1.tgz
        version:
            0.2.1
```

This issue is on Centos6
 - salt-call 2014.7.5 (Helium)
 - npm-1.3.6-5.el6

**Comment by rallytime** [comment]
Hrm...I thought we had fixed this one already. But perhaps it has regressed. See the discussion in #18647.

**Comment by vkurup** [comment]
I'm seeing the same behavior on salt 2015.5.0:

Relevant portion of my state file:
```
npm:
  pkg.installed

less@2.1.0:
  npm.installed:
    - require:
      - pkg: npm
```

Highstate output:
```
[127.0.0.1:2222] out:   Name: less@2.1.0 - Function: npm.installed - Result: Changed
```

Minion logs:
```
2015-06-22 16:25:09,928 [salt.state       ][INFO    ][9411] Completed state [npm] at time 16:25:09.927735
2015-06-22 16:25:09,930 [salt.state       ][INFO    ][9411] Running state [less@2.1.0] at time 16:25:09.930404
2015-06-22 16:25:09,931 [salt.state       ][INFO    ][9411] Executing state npm.installed for less@2.1.0
2015-06-22 16:25:09,932 [salt.loaded.int.module.cmdmod][INFO    ][9411] Executing command 'npm list --silent --json --global' in directory '/root'
2015-06-22 16:25:10,158 [salt.loaded.int.module.cmdmod][INFO    ][9411] Executing command 'npm install --silent --json --global "less"' in directory '/root'
2015-06-22 16:25:14,427 [salt.minion      ][INFO    ][7723] User sudo_vagrant Executing command saltutil.find_job with jid 20150622162514421129
2015-06-22 16:25:14,437 [salt.minion      ][INFO    ][10364] Starting a new job with PID 10364
2015-06-22 16:25:14,450 [salt.minion      ][INFO    ][10364] Returning information for job: 20150622162514421129
2015-06-22 16:25:19,479 [salt.minion      ][INFO    ][7723] User sudo_vagrant Executing command saltutil.find_job with jid 20150622162519468029
2015-06-22 16:25:19,488 [salt.minion      ][INFO    ][10375] Starting a new job with PID 10375
2015-06-22 16:25:19,500 [salt.minion      ][INFO    ][10375] Returning information for job: 20150622162519468029
 -- Elided npm list results -- 
2015-06-22 16:25:24,074 [salt.state       ][INFO    ][9411] {'new': ['less@2.1.0'], 'old': []}
2015-06-22 16:25:24,074 [salt.state       ][INFO    ][9411] Completed state [less@2.1.0] at time 16:25:24.074150
```

Actual result on the server:
```
vagrant@vagrant-ubuntu-trusty-64:~$ sudo salt-call --local npm.list less
[INFO    ] Executing command 'npm --version' in directory '/home/vagrant'
[INFO    ] Executing command 'npm list --silent --json --global "less"' in directory '/home/vagrant'
[ERROR   ] Invalid outputter mixed specified, fall back to nested
local:
    ----------
    less:
        ----------
        from:
            less@
        version:
            2.5.1
```

**Comment by zerthimon** [comment]
Doen't work in 2015.5.3

**Comment by rallytime** [comment]
I was able to reproduce this error, and I've fixed it with PR #29715 on the 2015.5 branch. The fix will propagate to the 2015.8 and develop branch within the next couple of days. Here's the results of the fix:
```
# salt-call --local state.sls npm
local:
----------
          ID: npm
    Function: pkg.installed
      Result: True
     Comment: Package npm is already installed
     Started: 18:39:16.288579
    Duration: 876.885 ms
     Changes:
----------
          ID: less@2.1.0
    Function: npm.installed
      Result: True
     Comment: Package(s) 'less@2.1.0' successfully installed
     Started: 18:39:17.168496
    Duration: 5975.98 ms
     Changes:
              ----------
              new:
                  - less@2.1.0
              old:

Summary for local
------------
Succeeded: 2 (changed=1)
Failed:    0
------------
Total states run:     2
#
# salt-call --local npm.list less
local:
    ----------
    less:
        ----------
        from:
            less@2.1.0
        version:
            2.1.0
#
# salt-call --local test.versions_report
local:
               Salt: 2015.5.6-132-gf606c23
             Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
             Jinja2: 2.7.3
           M2Crypto: 0.21.1
     msgpack-python: 0.4.4
       msgpack-pure: Not Installed
           pycrypto: 2.6.1
            libnacl: Not Installed
             PyYAML: 3.11
              ioflo: Not Installed
              PyZMQ: 14.0.1
               RAET: Not Installed
                ZMQ: 4.0.4
               Mako: 0.9.1
            Tornado: 4.2
            timelib: 0.2.4
           dateutil: 2.4.2
#
# cat /srv/salt/npm.sls
npm:
  pkg.installed

less@2.1.0:
  npm.installed:
    - require:
      - pkg: npm
```
